src/features/game/actions/buySFL.ts

```diff
@@ -0,0 +1,47 @@
+import { wallet } from "lib/blockchain/wallet";
+import { CONFIG } from "lib/config";
+import { ERRORS } from "lib/errors";
+
+const API_URL = CONFIG.API_URL;
+
+type Options = {
+  farmId: number;
+  token: string;
+  transactionId: string;
+  matic: string;
+  amountOutMin: string;
+};
+
+export async function buySFL({
+  farmId,
+  token,
+  transactionId,
+  matic,
+  amountOutMin,
+}: Options) {
+  const response = await window.fetch(`${API_URL}/buy-sfl/${farmId}`, {
+    method: "POST",
+    headers: {
+      "content-type": "application/json;charset=UTF-8",
+      Authorization: `Bearer ${token}`,
+      "X-Transaction-ID": transactionId,
+    },
+  });
+
+  if (response.status === 429) {
+    throw new Error(ERRORS.TOO_MANY_REQUESTS);
+  }
+
+  if (response.status >= 400) {
+    throw new Error(ERRORS.SYNC_SERVER_ERROR);
+  }
+
+  const transaction = await response.json();
+
+  // Do contract call
+  const receipt = await wallet
+    .getBuySFL()
+    .buySFL({ ...transaction, matic, amountOutMin });
```

Comment:
how does this contract deals with slippage?

Reply:
We do set slippage to 1%.

The purpose of this feature is to help folks who aren't really familiar with crypto swap matic for SFL easily. We chose to leave off a slippage setting because it's confusing. If folks what a more fine grained swap they can use quickswap directly. That's the thinking. We don't want to be a complete alternative to quickswap rather a simple alternative.